### PR TITLE
OHM-248: Move auditing into authentication middleware

### PR DESCRIPTION
**Pull request by hnnesv**
Add auditing exemption paths

OHM-248

### CODE REVIEW

File: src/api/authentication.coffee [hunk 1]
```diff
@@ -3,6 +3,23 @@ crypto = require 'crypto'
...
+config.auditing = config.get('auditing')
+atna = require 'atna-audit'
+auditing = require '../auditing'
+os = require 'os'
+himSourceID = config.auditing.auditEvents.auditSourceID
+
+# will NOT audit any successful logins on the following paths (specified as regex patterns)
+# only 'noisy' endpoints should be included, such as heartbeats or endpoints that get polled
+auditingExemptPaths = [
+  /\/transactions/
```
Comment (rcrichton) [inline src/api/authentication.coffee]: Should transaction be included here? They may be the most important to audit.

**Comment by rcrichton** [comment]
@hnnesv looks good. Just a question above.

File: src/api/authentication.coffee (hunk 1 above)
Comment (hnnesv) [inline src/api/authentication.coffee]: It is one the endpoints that gets polled though by console. What do you think - should we allow it then regardless? It is quite noisy

Comment (rcrichton) [inline src/api/authentication.coffee]: Yeah, the only problem is that we are possibly disclosing PHI when we query for transactions. Does the polling use a simple view of the transactions as a request param that we could match on?

Comment (hnnesv) [inline src/api/authentication.coffee]: Yup it does - yeah good idea. We can add a check to only exempt if on a simple view.

**Comment by hnnesv** [comment]
@rcrichton addressed the /transactions case

**Comment by rcrichton** [comment]
Nice looks good :+1:

**Pull request merged**